frontend/src/domain/shared_slices/Mission.ts

```diff
@@ -0,0 +1,55 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+import type { MissionFormValues } from '../../features/SideWindow/MissionForm/MainForm/types'
+import type { Mission } from '../types/mission'
+import type { PayloadAction } from '@reduxjs/toolkit'
+
+export interface BeaconMalfunctionState {
+  draftFormValues: MissionFormValues | undefined
+  editedMission: Mission | undefined
+}
+const INITIAL_STATE: BeaconMalfunctionState = {
+  draftFormValues: undefined,
+  editedMission: undefined
+}
+
+const missionSlice = createSlice({
```

Comment:
This slice is not only dedicated to edition, it's also about creation and may be about listing later on. It's a common practice to have your CRUD-related slices following your entity names. You will see that in almost any example about Redux slices.

That's why it's a strange architecture to use clean archi in a frontend app. Because we generally gather API services, Entity-related typings, slices and dispatchers together since not only most of them have exactly the same name, but they are closely related to each other.

Reply:
Ok for keeping it like this.

>  it's a strange architecture to use clean archi in a frontend app
It's only about dependency injection, the else remain the same.